ironfish/src/workerPool/tasks/decryptNotes.ts

```diff
@@ -225,6 +226,31 @@ export class DecryptNotesResponse extends WorkerMessage {
 
     return size
   }
+
+  /**
+   * Groups each note in the response by the account it belongs to. The
+   * `accounts` passed must be in the same order as the `accountKeys` in the
+   * `DecryptNotesRequest` that generated this response.
+   */
+  mapToAccounts(
+    accounts: ReadonlyArray<{ accountId: string }>,
+  ): Map<string, Array<DecryptedNote | null>> {
+    const decryptedNotesByAccount: Array<
+      [accountId: string, notes: Array<DecryptedNote | null>]
```

Comment:
`.map` is much slower in some cases than `for..of`. Not sure if this is one of those cases just checking to see if you are aware

Reply:
It's true that a for loop can outperform `.map()`, but in this very specific case I'm trying to get an array out of another array, so I'd expect the performance difference to be minimal. I would be more concerned if I was doing chains of `.map`/`.filter`, because those create a new array each time, and that would be better written as a for loop (at least until Node.js gains the ability to optimize these methods). Thanks for pointing it out though.